src/rubicon/objc/api.py

```diff
@@ -840,15 +846,22 @@ class or a metaclass, an instance of :class:`ObjCClass` or
                 # If an ObjCInstance already exists for the Objective-C object,
                 # reuse it instead of creating a second ObjCInstance for the
                 # same object.
-                return cls._cached_objects[object_ptr.value]
+                cached_obj = cls._cached_objects[object_ptr.value]
+
+                # If a cached instance was returned from a call such as `copy` or
+                # `mutableCopy`, we take ownership of an additional refcount. Release
+                # it here to prevent leaking memory, Python already owns a refcount from
+                # when the item was put in the cache.
+                if _returned_from_method.startswith(_OWNERSHIP_METHOD_PREFIXES):
+                    send_message(object_ptr, "release", restype=objc_id, argtypes=[])
```

Comment:
I worry a bit that the mental load of following when we retain and release is a bit much now since the reader needs to think though the `__new__` method being invoked multiple times with the same pointer.

Alternatives such as explicitly tracking a "Python refcount" might be easier to understand but would also add complexity.

Suggestions are welcome.

Reply:
I'm fine with this approach: if it's possible to deal with the situation immediately after the `copy`, that's definitely easier to follow than maintaining an extra refcount.

Reply:
I agree. An extra refcount starts to get into re-implementing garbage collection territory, and I'm not sure that's a game that is worth it. I'm comfortable with this as a known edge case, and documenting it with some related usage patterns and advice.

Reply:
Ok, then I'll stick with this :)